### DISCUSSION TITLE
Execution reverted for an unknown reason, better return

### DISCUSSION THREAD

**Discussion by joaofranciscoguarda** [wevm/viem#1579] [Idea / Feature Request]
### Suggestion: 
Returning the keccak256 or the already readable error message, such as Tendely does. Or returning the RPC response within the error message.

### Execution:
```
EstimateGasExecutionError: Execution reverted for an unknown reason.
Estimate Gas Arguments:
from: 0xF209c0F40B6B3317F09b33Ba89299d0C5f624b5E
to:0x4F927477D842F7d0ebE3E9804dfb29419Dd104ea
data:0x1d8de74c000000000000000000000000421a268deb037d25281e6c7d143c433bdfb9879c000000000000000000000000f209c0f40b6b3317f09b33ba89299d0c5f624b5e0000000000000000000000000000000000000000000000000000000065771e443b15a31c0059f9281df56342d6125964bd03863b9c1ef5a2f8c99b49b6b26c04
Details: execution reverted Version: viem@1.17.1
```

### Tendely:
![image](https://github.com/wevm/viem/assets/95258409/c33b3525-29b8-42ac-b6b8-844b771155f6)

Viem is very powerfull and can't let this pass. I'm using viem for a back-end project, the errors and solutions provided are very handy, but returning nothing is a problem, very hard to debug with nothing usefull in the logs. 

I'm using `estimateGas` to avoid using `simulateContract`, because I had to pass to much information (such as the ABI) when it's not needed to send all those information only to simulate (and because even with wrong data or addresses the simulate passes, so using .call( ) it's not safe enough for the situation), and to continue the process, having the needed gas is needed, so, it fits perfectly in the use case. In the execution I'm using `sendRawTransaction`

The error in case is related to someone that already minted an assets, so would be perfect to handle a already minted case, but I can't know if it's the case when the error is "blank"

Ps: It's not a rpc error, the logs show a try in 3 different rpc, same error in 3 tries.

**Comment by frolic** [comment]
You can use `getContractError` like `simulateContract` does under the hood:
https://github.com/wevm/viem/blob/5f350ecf63b2da861c8dcb60e77fbcae3e7192ab/src/actions/public/simulateContract.ts#L305-L314
